internal/taskrunner/preloadedtaskrunner.go

```diff
@@ -54,9 +54,9 @@ func (tr *PreloadedTaskRunner) Start() {
 	}
 }
 
-// StartAndWait starts running the tasks in the task runner and waits for them to complete.
-func (tr *PreloadedTaskRunner) StartAndWait() error {
-	tr.Start()
+// Wait calls .Wait() on the underlying waitgroup, allowing for e.g.
+// cleanup of existing tasks to happen.
+func (tr *PreloadedTaskRunner) Wait() error {
```

Comment:
Splitting out this logic so that we can `Wait()` on a taskrunner to finish all of its goroutines (e.g. during context cancellation) before tearing down the goroutine that spawned the taskrunner